### ISSUE TITLE
Android flags overlap with browser and os

### ISSUE THREAD

**Issue by xavadu** [lancedikson/bowser#160]
Hello, 

I would suggest to have different flags names for the os and browser in order to avoid overlap, or prefix them.

It happen that on a code like the following:

```
// Detect browser
if (bowser.android) {
 //...
} else if (bowser.chrome) {
 // ..
} } else if (bowser.firefox) {
 // ..
}
```

If an user is coming from an android device doesn't matter if it is using Firefox, Chrome or any other browser, the first if always will be true, because the same flag name is used to detect the browser (android native) and the os (android).

If the changes are accepted I could make the changes and do a PR.

Thanks,
J.

**Comment by lancedikson** [comment]
Hi, @xavadu. It looks like a feature for the next major release. I mean, now I'm working on the new version of the library. It will have convenient API and this suggestion looks like suitable for the new version. And suggested changes are so big that we need to upgrade the major version. It looks like unsuitable because of the new version is coming.

Btw, I think that in the next version there are no that flags for OS and browser name anymore. I mean, it would be look like

```
bowser.name
> Chrome
bowser.version
> 53.0
bowser.os
> macOS
// etc
```

**Comment by xavadu** [comment]
Hello, nice to see that the development never end :+1: 

Looks good the version 2, a suggestion for it also could be have a method as the following:
```
bowser.isBrowser('chrome');
> true
bowser.isOs('ios');
false
```

Regarding to the flag overlap of the current version, a quick and not bad fix could be to change the _android_ flag for the os, per example:
```
bowser.androidos 
// and
bowser.android
```

**Comment by lancedikson** [comment]
I don't think that we need to change something in the current version. It has not only browser and OS flags, but sometime it has a device flag: `ipod` and `iphone`. And it looks like an overlap too. So, let's work on the next version. I like your suggestions and I'll rely on them.

**Comment by arbesfeld** [comment]
Any suggested workarounds for the current version of the library?

**Comment by lancedikson** [comment]
@arbesfeld, now you still able to use `bowser.android` and all those flags described in docs. It's quite simple to add a new API. I'm gonna do it during the next week. It wouldn't be a major version. I'll just add some new methods, but the next major version will deprecate old stuff like flags.

**Comment by xavadu** [comment]
the android flag is used for two different things: android devices and android browser, check my suggestion to just only rename on of the flag, it will solve the issue meanwhile in an easy an elegant way.

**Comment by lancedikson** [comment]
Yeah, agree. Needs to be fixed. I'll deprecate using `bowser.android` flag and will remove them later in the next major release. Thanks, @xavadu for the suggestion :)

**Comment by xavadu** [comment]
But which ``bowser.android`` tag? the one for the OS or the one for the browser? is better just to change the name, so a functionality is not lost. Thanks to you!

**Comment by lancedikson** [comment]
Answering the question: 
I will add `.androidOs` and a new API to check that. 

Details:
Actually, I think, it'd be ok to don't change those flags, but add new functionality like `.isOs()` and `.isBrowser()`, that will fix that problem with overlapping flags. We need to support backward compatibility with the older versions.
And then we could add a flag like `.androidOs`, but I prefer to deprecate all those flags that have been there all the time. I think bowser should have more convenient API than it has now.
I mean, first of all, I'll fix the problem with overlapping using the new API and then will see, what we should do next with it :)

**Comment by xavadu** [comment]
Sound very good have a group for each flag as you describe ``.isBrowser().android`` or ``isOs().android``. Thanks!

**Comment by lancedikson** [comment]
No, it's not what I meant. I'd like to avoid flags at all and use only API like `bowser.isOs('android')` and `bowser.isBrowser('android')`. Smth like this :)

**Comment by xavadu** [comment]
aah okay, sound good too, but not sure if have advantages, I think that could be more easy to make a typo or mistake, but with a good error information in the console if the parameter is not expected could be good

**Comment by lancedikson** [comment]
Yeah, I thought about those additional symbols and possible mistakes if compare with flags, but, it may be more convenient for example to cover some format and trim things in methods, like loyal browser name parameter (`.isBrowser(' AnDrOiD ')`) or smth like that 😄

**Comment by adambyer** [comment]
Hi, do you know the status of this? I don't see either isBrowser or androidOs in the code or docs.

**Comment by lancedikson** [comment]
It's been fixed in 2.0.0-alpha.1
